### PR TITLE
hazelcast-cpp-client: add version 5.1.0

### PR DESCRIPTION
**Pull request by OzanCansel**
Specify library name and version:  **hazelcast-cpp-client/5.1.0**

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by uilianries — APPROVED** [review]

**Comment by OzanCansel** [comment]
Guys can someone review this PR. It passes all the checks.

**Comment by ihsandemir** [comment]
Can you please review and merge @uilianries ? We like to use it in projects using Conan.

File: recipes/hazelcast-cpp-client/all/patches/gcc_11_1_0.patch [hunk 1]
```diff
@@ -0,0 +1,40 @@
+diff --git a/hazelcast/include/hazelcast/client/protocol/ClientMessage.h b/hazelcast/include/hazelcast/client/protocol/ClientMessage.h
+index 462ea85b0..f2686506a 100644
+--- a/hazelcast/include/hazelcast/client/protocol/ClientMessage.h
++++ b/hazelcast/include/hazelcast/client/protocol/ClientMessage.h
+@@ -190,6 +190,9 @@ struct HAZELCAST_API is_trivial_entry_vector<
+  */
+ class HAZELCAST_API ClientMessage
+ {
++    template<typename T>
```
Comment (prince-chrismc) [inline recipes/hazelcast-cpp-client/all/patches/gcc_11_1_0.patch:9]: You should contribute this back upstream to the project

**Review by prince-chrismc — APPROVED** [review]

**Comment by prince-chrismc** [comment]
You can see the current status https://github.com/prince-chrismc/conan-center-index-pending-review/issues/1

The bottom chart shows the estimated time are is almost 2 weeks.

Holidays + backend bugs slowed us down.

File: recipes/hazelcast-cpp-client/all/patches/gcc_11_1_0.patch (hunk 1 above)
Comment (OzanCansel) [reply recipes/hazelcast-cpp-client/all/patches/gcc_11_1_0.patch:9]: Do you mean previous versions at conan or the hazelcast/hazelcast-cpp-client repo ?

Comment (prince-chrismc) [reply recipes/hazelcast-cpp-client/all/patches/gcc_11_1_0.patch:9]: To hazelcast/hazelcast-cpp-client repo :)

**Review by uilianries — APPROVED** [review]
LGTM

**Pull request merged**